src/main/java/ca/uhn/fhir/jpa/starter/service/HelperService.java

```diff
@@ -1568,19 +1646,23 @@ static String getValidURL(String invalidURLString) {
 
 	private String createKeycloakGroup(GroupRepresentation groupRep) {
 		RealmResource realmResource = FhirClientAuthenticatorService.getKeycloak().realm(appProperties.getKeycloak_Client_Realm());
-		List<GroupRepresentation> groups = realmResource.groups().groups(groupRep.getName(), 0, Integer.MAX_VALUE, true);
-
+		List<GroupRepresentation> groups = realmResource.groups().groups(groupRep.getName(), 0, Integer.MAX_VALUE, false);
+//
 		for (GroupRepresentation group : groups) {
-			if (group.getName().equals(groupRep.getName())) {
+			if (group.getName().equals(groupRep.getName()) && (groupRep.getAttributes().get("parent") == null || group.getAttributes().containsValue(groupRep.getAttributes().get("parent")))) {
 				return group.getId();
 			}
 		}
 		try {
 			Response response = realmResource.groups().add(groupRep);
 			return CreatedResponseUtil.getCreatedId(response);
 		} catch (WebApplicationException ex) {
-			logger.warn(ex.getLocalizedMessage());
-			return null;
+			logger.warn("Group with identical name found. Appending the parent group name to existing name.");
+			GroupResource parentGroupResource = realmResource.groups().group(groupRep.getAttributes().get("parent").get(0));
+			GroupRepresentation parentGroup = parentGroupResource.toRepresentation();
+			String parentName = parentGroup.getName();
+			groupRep.setName(groupRep.getName() + " " + parentName);
```

Comment:
Use another separator instead of " " . Invert it , i.e. parentName groupRep.getName() . Also The name based search in the above lines should work for such groups that have parent name appended to it.